workshops/views.py

```diff
@@ -215,6 +228,18 @@ class AirportUpdate(LoginRequiredMixin, UpdateViewContext):
     slug_url_kwarg = 'airport_iata'
     template_name = 'workshops/generic_form.html'
 
+
+@login_required
+def airport_delete(request, airport_iata):
+    """Delete specific airport."""
+    try:
+        airport = get_object_or_404(Airport, iata=airport_iata)
+        airport.delete()
+        messages.success(request, 'Airport was deleted successfully.')
+        return redirect(reverse('all_airports'))
+    except IntegrityError:
```

Comment:
Maybe catch the more-specific [`ProtectedError`][1] here?

[1]: https://docs.djangoproject.com/en/1.8/ref/exceptions/#django.db.models.ProtectedError